core/src/main/scala/io/qbeast/spark/index/QbeastColumns.scala

```diff
@@ -202,4 +213,12 @@ case class QbeastColumns(
    *   the cube to rollup column exists
    */
   def hasCubeToRollupColumn: Boolean = cubeToRollupColumnIndex >= 0
+
+  /**
+   * Returns whether the filename column exists.
+   *
+   * @return
+   *   the filename column exists
+   */
+  def hasFilenameColumn: Boolean = filenameColumnIndex >= 0
```

Comment:
There's no other option for this feature than using a column in the dataframe? Couldn't it be calculated on the fly and returned in the output file stats? 

If there's no other way around it, should we add another component that takes care of dividing the DataFrame into files based on its logic? In that way, we can accommodate different strategies to choose and distribute files.

Reply:
For example:

`OTreeManager` -> `FileManager` -> `DataWriter`

Reply:
> There's no other option for this feature than using a column in the dataframe?

In this case, yes. The file name must be present in all the rows of the DataFrame before storing it to the Parquet file when using Hudi. For Delta, this is unnecessary.

> If there's no other way around it, should we add another component that takes care of dividing the DataFrame into files based on its logic?

The logic that handles this is in the `RollupDataWriter`. If the column is not set, it will work as before, without additional overhead.

Reply:
> The logic that handles this is in the RollupDataWriter

Yeaaa, but I think we should have an interface between the RollUp and the final DataWriter. Maybe for another PR, if we agree 🐒 


What I am not convinced from a logical perspective, is that we use the `getRollupCubeIdUDF` to return something that is not a CubeId... 
Also, the `fileName` contains the string for the `cubeId` somewhere? Would it be a good practise to save it as cubeID + timestamp or cubeID + UUID ?

Reply:
> What I am not convinced from a logical perspective, is that we use the getRollupCubeIdUDF to return something that is not a CubeId...

In practice, the UUID is a unique identifier for the Cube. That's why I included it here. Another option I considered was creating a separate column, `cubeToRollupUUID`, but then the original `cubeToRollup `column would go unused, as its only purpose is to create a different data writer for each diferent `cubeToRollup`, which could be done with the `cubeToRollupUUID`.

> Also, the fileName contains the string for the cubeId somewhere? Would it be a good practise to save it as cubeID + timestamp or cubeID + UUID ?

As of today the filename does not contain the cube ID. I'm not certain about Delta, but Hudi is very strict with its filename format, so including it is not possible. The approach I'm following now is to make the filename exactly match the format expected by each underlying table format. In Delta, it's `UUID.parquet`; in Hudi, it's `UUID_TOKEN_TIMESTAMP.parquet`; and in Iceberg, it appears to be `UUID.parquet`.

Reply:
>  the UUID is a unique identifier for the Cube..

mmm, I think it's important not to merge the concepts of Cube Identifier with File Identifier. The CubeID is not (or should not be) a UUID. UUID is unique, while the Cube ID can have the same value in multiple use cases. 

A different data writer should be created for a different File. In this case (the default), each file belongs to the rollup cube, but for example, if we want to turn RollUp off (issue #460 ), this casuistic would not be applied. 

> The approach I'm following now is to make the filename exactly match the format expected by each underlying table format.

Then sounds good to add a UUID!! 

Just make sure is referenced as `fileUUID` and it does not interfere with the Cube ID concept. Even if the mapping is 1:1, they are created through an entirely different process and have different meanings. And allows the code to be more flexible.

Reply:
So, in your opinion, how would you proceed?

- Keep the `rollupCubeID` column and add another one for the `cubeUUID`?
- Remove the `rollupCubeID` column and create a more generic column, such as `partitionUUID`, which identifies the destination partition (parquet file) for the data? In the case of a rollup, `cubeToRollup` would populate the `partitionUUID`, but if the rollup is deactivated, a different logic would populate the `partitionUUID` for each row.
- Other ideas?

Reply:
After small discussion, we agreed to:
- using `fileUUID` as a file identifier, independently of `rollupCubeID`.
- For the DataWriter, the only important column name should be fileID, which would create a writer per group, 
- In other PR's, we should think about decoupling the `RollUp` from the `DataWriter` code.